torch/_meta_registrations.py

```diff
@@ -206,6 +223,61 @@ def meta_linalg_eigh(self, uplo="L"):
     return (values, vectors)
 
 
+# From c10/util/strides.h
+def contiguous_strides(sizes):
+    """Computes the contiguous strides of a tensor, given its sizes."""
+    # With this intialisation we get the case dim == 0 or 1 right
+    dims = len(sizes)
+    strides = [1] * dims
+
+    i = dims - 2
+    while i >= 0:
+        # Strides can't be 0 even if sizes are 0.
+        strides[i] = strides[i + 1] * max(sizes[i + 1], 1)
+        i -= 1
+
+    return strides
```

Comment:
I feel like we've already got ac opy of this function floating around in python

Reply:
yeah you are right `make_contiguous_strides_for` in `torch/_prims_common/__init__.py`